### PR TITLE
Cr87 retrieve case by case ref

### PR DESCRIPTION
**Pull request by PeterBochel**
# Motivation and Context
Implemented the the contact centre get case by case reference endpoint. 
This is very similar to the get case by case id endpoint, and mostly delegates it's work to the case service.

# How to test?
Unit tests have been added to test the new code. 

# Links
See: https://collaborate2.ons.gov.uk/jira/browse/CR-87

### CODE REVIEW

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java [hunk 1]
```diff
@@ -45,53 +47,92 @@ public void initMocks() {
...
 
   @Test
-  public void testGetCaseByCaseId_withNoCaseDetails() throws Exception {
+  public void testGetCaseByCaseRef_withNoCaseDetails() throws Exception {
+    doTestGetCaseByCaseRef(false);
+  }
+
+  @Test
+  public void testGetCaseByCaseRef_nonHouseholdCase() throws Exception {
+    UniquePropertyReferenceNumber testCaseRef = new UniquePropertyReferenceNumber("88234544");
```
Comment (philwhiles) [inline src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java]: What is a UPRN doing here?

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java [hunk 2]
```diff
@@ -45,53 +47,92 @@ public void initMocks() {
...
     CaseRequestDTO requestParams = new CaseRequestDTO(caseEvents);
     CaseDTO results = caseService.getCaseById(uuid, requestParams);
 
     verifyCase(results, caseEvents);
   }
 
-  @Test
-  public void testGetCaseByCaseId_nonHouseholdCase() throws Exception {
+  private void doTestGetCaseByCaseRef(boolean caseEvents) throws Exception {
+    UniquePropertyReferenceNumber testCaseRef = new UniquePropertyReferenceNumber("88234544");
```
Comment (philwhiles) [inline src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java]: UPRN...

Comment (PeterBochel) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java]: Fixed

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java (hunk 1 above)
Comment (PeterBochel) [reply src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java]: Fixed

**Pull request merged**